### PR TITLE
[PHP-17] create DB hometask

### PR DESCRIPTION
**Pull request by oleksiivv**

### CODE REVIEW

File: src/db/QueryBuilder.php [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+        global $pdo;
+        $sth = $pdo->prepare($query);
+        $sth->setFetchMode(\PDO::FETCH_ASSOC);
+        
+        foreach ($params as $key => $value) {
+            if ($key === ':startIndex' || $key === ':itemsCount') {
+                $sth->bindValue($key, $value, \PDO::PARAM_INT);
+            }
+            else {
```
Comment (yderevianko-cartera) [inline src/db/QueryBuilder.php:76]: else should be on same line as }
example:
} else {

File: src/db/QueryBuilder.php [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            }
+            
+            $params[':firstLetterName'] = $this->queryParams["filter-first-letter-name"];
+        }
+        if (isset($this->queryParams["filter-first-letter-state"])) {
+           
+            if(strlen($queryFilter) === 0) {
+                $queryFilter = " WHERE s.name LIKE CONCAT(:firstLetterState, '%')";
+            }
+            else{
```
Comment (yderevianko-cartera) [inline src/db/QueryBuilder.php:47]: else should be on same line as }
example:
} else {

File: src/db/QueryBuilder.php [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+        $queryFilter = "";
+        $querySort = "";
+        $queryLimit = "";
+        
+        if (isset($this->queryParams["filter-first-letter-name"])) {
+           
+            if (strlen($queryFilter) === 0) {
+                $queryFilter = " WHERE a.name LIKE CONCAT(:firstLetterName, '%')";
+            }
+            else {
```
Comment (yderevianko-cartera) [inline src/db/QueryBuilder.php:36]: else should be on same line as }
example:
} else {

File: src/db/functions.php [hunk 4]
```diff
@@ -0,0 +1,126 @@
...
+/**
+ * @param $param
+ * @param $value
+ * @return string
+ */
+function addParamToURL($param, $value): string
+{
+    $query = $_GET;
+
+    if($param !== "page") {
```
Comment (yderevianko-cartera) [inline src/db/functions.php:78]: please add space after if

File: src/db/functions.php [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+    return $uniqueLetters;
+}
+
+/**
+ * @param PDO $pdo
+ * @return int
+ */
+function countRows(PDO $pdo): int
+{
+
```
Comment (yderevianko-cartera) [inline src/db/functions.php:27]: not needed empty line

File: src/db/functions.php [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+require_once './QueryBuilder.php';
+
+
+/**
+ * @param PDO $pdo
+ * @return array
+ */
+function getUniqueFirstLetters(PDO $pdo): array
+{
+
```
Comment (yderevianko-cartera) [inline src/db/functions.php:12]: not needed empty line

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/db/QueryBuilder.php [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+            $params[':startIndex'] = $this->queryParams["limit-start-index"];
+            $params[':itemsCount'] = $this->queryParams["limit-items-count"];
+        }
+
+        $query = "SELECT a.name, a.code, a.address, a.timezone, c.name AS city, s.name AS state FROM airports AS a 
+                    INNER JOIN cities AS c ON a.city_id = c.id 
+                    INNER JOIN states AS s ON a.state_id = s.id"
+                    . $queryFilter . $querySort . $queryLimit;
+
+        global $pdo;
```
[the remark this answers is not in the document]
Comment (oleksiivv) [reply src/db/QueryBuilder.php]: fixed

File: src/db/QueryBuilder.php [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+    {
+        $this->queryParams[$command] = $param;
+    }
+
+    /**
+     * @return array
+     */
+    public function buildQuery(): array
+    {
+
```
[the remark this answers is not in the document]
Comment (oleksiivv) [reply src/db/QueryBuilder.php]: fixed

File: src/db/QueryBuilder.php (hunk 1 above)
Comment (oleksiivv) [reply src/db/QueryBuilder.php]: fixed

File: src/db/QueryBuilder.php (hunk 2 above)
Comment (oleksiivv) [reply src/db/QueryBuilder.php]: fixed

File: src/db/QueryBuilder.php (hunk 3 above)
Comment (oleksiivv) [reply src/db/QueryBuilder.php]: fixed

File: src/db/functions.php (hunk 4 above)
Comment (oleksiivv) [reply src/db/functions.php]: fixed

File: src/db/functions.php (hunk 5 above)
Comment (oleksiivv) [reply src/db/functions.php]: fixed

File: src/db/functions.php (hunk 6 above)
Comment (oleksiivv) [reply src/db/functions.php]: fixed

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**